arch/Kconfig

```diff
@@ -196,6 +197,25 @@ config SRAM_BASE_ADDRESS
 	  /chosen/zephyr,sram in devicetree. The user should generally avoid
 	  changing it via menuconfig or in configuration files.
 
+# Workaround for not being able to have commas in macro arguments
+DT_CHOSEN_Z_DRAM := zephyr,dram
```

Comment:
Any new chosen option needs to be documented in the list at the bottom of this page: https://docs.zephyrproject.org/latest/build/dts/api/api.html#devicetree-chosen-nodes

I would like to see a strong justification for why you need this option at all. The commit logs in the PR in general are too short and lack justification given that they are adding new chosen nodes and touching arch/Kconfig.

Reply:
The Corstone-300 reference platform has beside TCM also SRAM and DRAM. We would like to control which memory the TFLM model is and arena are mapped to. How do we best do this?

Reply:
@carlocaione can you advise here?

Reply:
not sure if this exactly fits what you are trying to do but you can leverage the `zephyr,memory-region` compatible to create the DRAM region and allocate the proper sections in there.

See for example: https://github.com/zephyrproject-rtos/zephyr/blob/0c053190b79dd1f121be6e71e2d4fb2e2251fecb/boards/arm/stm32f429i_disc1/stm32f429i_disc1.dts#L26-L31 

and then:

https://github.com/zephyrproject-rtos/zephyr/blob/0c053190b79dd1f121be6e71e2d4fb2e2251fecb/drivers/memc/memc_stm32_sdram.ld#L19-L29

Reply:
> I reverted the changes to arch/Kconfig and replaced that with a custom linker script in the TFML Ethos-U sample application, similar to the memc example above.

But the problematic `zephyr,dram` chosen node still remains and therefore my comments about it are not yet addressed.

Reply:
I have removed the `zephyr,dram` from the `chosen` section.